crates/precompile/src/secp256k1.rs

```diff
@@ -50,11 +50,11 @@ fn ec_recover_run(i: &[u8], target_energy: u64, network: Network) -> PrecompileR
     let mut sig = [0u8; 171];
     msg[0..32].copy_from_slice(&input[0..32]);
     sig[0..171].copy_from_slice(&input[96..32 * 3 + 171]);
-    let out = ecrecover(&sig, &msg, network)
-        .map(Vec::from)
-        .unwrap_or_default();
-
-    Ok((ECRECOVER_BASE, out))
+    let out = ecrecover(&sig, &msg, network).map(Vec::from);
+    if out.is_err() {
+        return Err(Error::EcrecoverBadData);
+    }
+    Ok((ECRECOVER_BASE, out.unwrap()))
```

Comment:
```suggestion
    let out = ecrecover(&sig, &msg, network)
        .map(Vec::from)
        .map_err(|_| Error::EcrecoverBadData)?;
    Ok((ECRECOVER_BASE, out))

```

Reply:
We should probably check the error we got from ed448 lib and report it to the user, now we only report one error no matter what happens inside ed448

Reply:
We could implement:

```rust
impl From<LibgoldilockErrors> for Error {}
``` 
Then we could just use ? for any error without the map_err() call